src/main/kotlin/com/ebay/plugins/metrics/develocity/MetricsForDevelocitySettingsPlugin.kt

```diff
@@ -68,45 +64,53 @@ internal class MetricsForDevelocitySettingsPlugin : MetricsForDevelocityPlugin<S
         }
         settings.plugins.withId("com.gradle.develocity") {
             val gradleExt = settings.extensions.getByType(DevelocityConfiguration::class.java)
-            settings.gradle.afterProject { project ->
-                project.plugins.withId("com.ebay.metrics-for-develocity") {
+            val server = gradleExt.server
+            val accessKey = gradleExt.accessKey
+
+            settings.gradle.lifecycle.afterProject { project ->
+                project.plugins.withType(MetricsForDevelocityPlugin::class.java) {
                     project.extensions.findByType(MetricsForDevelocityExtension::class.java)?.let { ext ->
                         with(ext) {
-                            develocityServerUrl.convention(gradleExt.server)
-                            develocityAccessKey.convention(gradleExt.accessKey)
+                            develocityServerUrl.convention(server)
+                            develocityAccessKey.convention(accessKey)
                         }
                     }
                 }
                 // Configure tasks wanting to consume the Develocity configuration:
                 project.tasks.withType(DevelocityConfigurationInputs::class.java).configureEach { task ->
                     // `convention` to allow for possible override by the property value
-                    task.develocityServerUrl.convention(gradleExt.server)
+                    task.develocityServerUrl.convention(server)
                 }
             }
         }
 
         // Look for task names that have a datetime or duration suffix and feed those into a property that the
         // plugin can consume in order to pro-actively create the consumable configurations.
-        settings.gradle.lifecycle.beforeProject { project ->
-            if (project.parent == null) {
-                val requestedTimeSpecs = mutableListOf<String>()
-                project.gradle.startParameter.taskRequests.forEach { taskRequest ->
-                    taskRequest.args.forEach { taskName ->
-                        DATETIME_SUFFIX_PATTERN.matchEntire(taskName)?.let { matchResult ->
-                            matchResult.groups[1]?.value?.let {
-                                requestedTimeSpecs.add(it)
-                            }
-                        }
-                        DURATION_SUFFIX_PATTERN.matchEntire(taskName)?.let { matchResult ->
-                            matchResult.groups[1]?.value?.let {
-                                requestedTimeSpecs.add(it)
-                            }
-                        }
+        val requestedTimeSpecs = mutableListOf<String>()
+        settings.gradle.startParameter.taskRequests.forEach { taskRequest ->
```

Comment:
So this is the one, that can't use `settings.gradle.lifecycle`?

Reply:
It was primarily the develocity / gradle enterprise extension object, but all of them were attempting to serialize types incompatible with the configuration cache, once I had them all switched over.